examples/src/androidTest/java/com/mapbox/navigation/examples/activity/TripServiceActivityKtTest.kt

```diff
@@ -84,7 +84,8 @@ class TripServiceActivityKtTest :
         }
     }
 
-    private inline fun awaitForView(resName: String, timeout: Long = 1000) {
+    // Increased timeout to 10.000 ms to check if it's enough for low power devices
```

Comment:
Not sure if this is the right solution. Wondering if there's something that we run in the background that causes sometimes issues. Have we checked thoroughly the logs in Firebase? How do we know this is only happening in low-end devices? Asking because in the past we've seen this flakiness in all of the devices we're using to run the UI tests in Firebase 🤔 Have we checked the videos of previous failing runs? Another idea is to explore if using [Android Test Orchestrator](https://developer.android.com/training/testing/junit-runner.html#using-android-test-orchestrator) this may increase the time of the run but it helps with false positives. I believe we should dig deeper to find the root cause of this flakiness.

Reply:
I like your Orchestrator idea 👍Added it 
Let me describe why I increased timeout to 10sec. Made a round of tests, assumed that our tests depends on Internet connection(if it has and its quality) and device performance. 
Have a look at this [line](https://github.com/mapbox/mapbox-navigation-android/blob/master/examples/src/main/java/com/mapbox/navigation/examples/core/TripServiceActivityKt.kt#L77): maps init async(it require good internet connection, and takes processor time), and in case if maps hasn't inited Notification Service cannot run(that takes processor time). 
I suggest run step by step, and check if everything work well with this approach because from my perspective of view we need to wait more for view appearance in cases what don't depends on us(device performance and internet connection)

Reply:
Still unsure about how this would help solving the flakiness 😬

Wondering if we could depend on something else to wait for the map to be loaded like [Espresso idling resources](https://developer.android.com/training/testing/espresso/idling-resource) or `CountDownLatch` (cc'ing @LukasPaczos as he has been recently working on performance metrics) instead of increasing the timeout that seems like a temporary workaround that can fail eventually.

In any case, do we have evidence of those assumptions you're mentioning? Did you see that from Firebase logs? I mean, if we need to wait 10 seconds to load a map we have a different problem 😅 (assuming there are no CI / infrastructure issues).

Reply:
As @Guardiola31337 mentioned, the robust solution would be to use a latch or an idling resource and wait for the `OnStyleLoaded` or `OnMapReady` callback (depending on the use case) and only then resume.